static/src/javascripts/projects/common/modules/commercial/reader-revenue-banner.js

```diff
@@ -0,0 +1,33 @@
+// @flow
+
+import type { Banner } from 'common/modules/ui/bannerPicker';
+import { fetchBannerData, renderBanner } from 'common/modules/commercial/contributions-service';
+
+
+const messageCode = 'reader-revenue-banner';
+
+let data = null;
+
+const show = () => renderBanner(data);
+
+const canShow = (): Promise<boolean> => {
+    const enabled = false;
+
+    // Temporarily disable banner while service is developed to serve actual banner and logic
+    if(!enabled) {
+        return Promise.resolve(false);
+    }
+
+    return fetchBannerData()
+        .then(moduleData => {
+            data = moduleData;
+            return data !== null;
```

Comment:
Can you lose the `let data` defined higher up and convert this to `return moduleData !== null` (or perhaps `return !!moduleData`)? I'm not sure what purpose `data` serves (though I might well be missing something!)

Reply:
We're implementing the `Banner` type here, which looks like:
```
type Banner {
  canShow: () => Promise<boolean>,
  show: () => Promise<boolean>
}
```
So without changing that contract, we can't pass data from the output of `canShow` into `show` (like we would want to for automat decisions, where the first request gives us a decision and some data).